provider/marathon/marathon_test.go

```diff
@@ -1801,3 +1807,33 @@ func TestParseIndex(t *testing.T) {
 		})
 	}
 }
+
+func TestMarathonGetBasicAuth(t *testing.T) {
+	provider := &Provider{}
+
+	applications := []struct {
+		application marathon.Application
+		expected    []string
+	}{
+		{
+			application: marathon.Application{
+				Labels: &map[string]string{}},
+			expected: []string{},
+		},
+		{
+			application: marathon.Application{
+				Labels: &map[string]string{
+					"traefik.frontend.auth.basic": "user:password",
+				},
+			},
+			expected: []string{"user:password"},
+		},
+	}
+
+	for _, a := range applications {
```

Comment:
Could you please turn this into a group of parallelized sub-tests given proper names? See some of the other Marathon tests where we already employ those.